### DISCUSSION TITLE
Is it possible to add a .editorconfig file outside the hierarchy of the project with file path globs?

### DISCUSSION THREAD

**Discussion by paulomorgado** [dotnet/roslyn#74575] [Q&A]
I have a few analyzers added via `CustomAfterMicrosoftCommon` environment variable and I'd like to add settings in editor config for `[*tests.cs]`, but it seems to be ignored, although a binlog shows that editorconfig being fed to the compiler..

What am I doing wrong?

**Comment by 333fred** [comment]
`.editorconfig` files apply to any files _underneath_ them. The file needs to actually be above the files you're hoping to influence. You may be looking for a `.globalconfig` instead: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#global-analyzerconfig.

**Reply by paulomorgado** [reply]
@333fred,

> `.editorconfig` files apply to any files _underneath_ them. The file needs to actually be above the files you're hoping to influence. You may be looking for a `.globalconfig` instead: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#global-analyzerconfig.

According to [the docs](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#format):

> Unlike EditorConfig files, which must have section headers, such as [*.cs], to identify the applicable files and folders, global AnalyzerConfig files don't have section headers.

So, my only option is to place the .editorconfig file in a common root for my projects.

**Reply by 333fred** [reply]
Yes, if you need section headers, then you need to place the file in a common root.
